### PR TITLE
Deploying PostgreSQL as a StatefulSet in Kubernetes

### PR DESCRIPTION
**Pull request by MariamFahmy98**

### CODE REVIEW

File: k8s/postgresql.yaml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+    matchLabels:
+      app: postgresql-db
+  template:
+    metadata:
+      labels:
+        app: postgresql-db
+    spec:
+      containers:
+        - name: postgresql-db
+          image: postgres:12.8
```
Comment (KhaledEmaraDev) [inline k8s/postgresql.yaml]: Why not use the latest version like `14.1-alpine3.14`?

File: k8s/postgresql.yaml [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+          ports:
+            - containerPort: 5432
+              name: http
```
Comment (KhaledEmaraDev) [inline k8s/postgresql.yaml]: PostgreSQL doesn't use HTTP.

File: k8s/postgresql-configmap.yaml [hunk 3]
```diff
@@ -0,0 +1,12 @@
...
+kind: ConfigMap
+metadata:
+  name: postgres-db-config
+  labels:
+    app: postgresql-db
+data:
+  POSTGRES_DB: gorm
+  POSTGRES_USER: postgres
+  POSTGRES_PASSWORD: postgres-test
+  PGDATA: /var/lib/postgresql/data/pgdata
```
Comment (KhaledEmaraDev) [inline k8s/postgresql-configmap.yaml]: Newline. :sweat_smile:

File: k8s/postgresql.yaml [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+  volumeClaimTemplates:
+    - metadata:
+        name: postgresql-db-disk
+      spec:
+        accessModes: ["ReadWriteOnce"]
+        resources:
+          requests:
+            storage: 1Gi
+
+
```
Comment (KhaledEmaraDev) [inline k8s/postgresql.yaml:68]: Newline. :grin:

**Review by KhaledEmaraDev — APPROVED** [review]

File: k8s/postgresql.yaml (hunk 2 above)
Comment (Hanaffi) [reply k8s/postgresql.yaml]: name: postgresdb
or just remove that line.

Not very sure tbh.

**Review by Hanaffi — CHANGES_REQUESTED** [review]

File: k8s/postgresql.yaml (hunk 2 above)
Comment (MariamFahmy98) [reply k8s/postgresql.yaml]: Yes, I have added it to try the DNS name by using the port name. (Just a dummy name)
So I think it is better to be removed for now.

**Pull request merged**